lexibank_lieberherrkhobwa.py

```diff
@@ -32,61 +32,56 @@ def cmd_install(self, **kw):
-                    ID=concept['CONCEPTICON_ID'],
-                    Name=concept['ENGLISH'],
-                    Concepticon_ID=concept['CONCEPTICON_ID'],
-                    Concepticon_Gloss=concept['CONCEPTICON_GLOSS'],
+                    ID=concept.id,
+                    Name=concept.english,
+                    Concepticon_ID=concept.concepticon_id,
+                    Concepticon_Gloss=concept.concepticon_gloss,
                 )
 
             # Read raw languages and append it
             langs = {}
             for language in self.languages:
-                langs[language['Name']] = language['Source']
+                langs[language["Name"]] = language["Source"]
                 ds.add_language(
-                    ID=language['Name'],
-                    Name=language['Name'],
-                    Glottocode=language['Glottocode'],
-                    Glottolog_Name=language['Glottolog_Name'],
+                    ID=language["Name"],
+                    Name=language["Name"],
+                    Glottocode=language["Glottocode"],
+                    Glottolog_Name=language["Glottolog_Name"],
                 )
 
             # iterate over the source adding lexemes and collecting cognates
-            modify = {"N": "N/ɴ", "’": "+"}
             for row in data[2:]:
                 lid = slug(row[0])
                 if lid in langs:
                     for cid in range(1, len(row), 2):
                         form = row[cid]
-                        if not form or (form == 'NA'):
+                        if not form or (form == "NA"):
                             continue
                         ipa = form
-                        ipa = ipa.replace('- ', '-').replace(' ', '_')
+                        ipa = ipa.replace("- ", "-").replace(" ", "_")
                         ipa = strip_brackets(ipa).strip()
 
                         if ipa:
                             # split ipa and remove final/initial underscores if any
                             # (without regular expressions)
-                            ipa = split_text(ipa, '~/~⪤,')[0]
+                            ipa = split_text(ipa, "~/~⪤,")[0]
 
-                            if ipa[0] == '_':
+                            if ipa[0] == "_":
                                 ipa = ipa[1:].strip()
-                            if ipa[-1] == '_':
+                            if ipa[-1] == "_":
                                 ipa = ipa[:-1].strip()
 
-                            tks = [tok for tok in ipa2tokens(ipa) if tok != '_']
-                            tks = [modify.get(x, x) for x in tks]
-
-                            ipa = split_text(ipa, '~/~⪤,')[0]
-                            tks = self.tokenizer(None, '^%s$' % ipa)
-                            tks = self.tokenizer(None, ipa, column='IPA')
+                            ipa = split_text(ipa, "~/~⪤,")[0]
+                            tks = self.tokenizer(None, ipa, column="IPA")
```

Comment:
This should not be done explicit, but it will be tokenized, if you provide a good form, so paste `ipa` as form, and have the tokenizer tokenize it properly automatically.

Reply:
See update.